Core/HistoManager.cpp

```diff
@@ -27,29 +27,90 @@ HistoManager::HistoManager() : fMakeControlHisto(true)
 
 HistoManager::~HistoManager() {}
 
+void HistoManager::createHistogramWithAxes(TObject* object, TString xAxisName, TString yAxisName, TString zAxisName) 
+{ 
+  TClass *cl = object->IsA();
+  if (cl->InheritsFrom("TH1D"))
+  {
+    TH1D* tempHisto = dynamic_cast<TH1D*>(object);
```

Comment:
I think that you could use static_cast instead of dynamic_cast if you check the type anyway. 
That should be faster. dynamic_cast is used to check the type at runtime, but you are kind of already doing it